src/dataflow-types/src/types.rs

```diff
@@ -470,17 +470,20 @@ impl Envelope {
 
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub enum SourceConnector {
```

Comment:
IMO at this point this type should just be called `Source`? its confusing to me to have a type called

 enum FooConnector with variant BarFooDesc with subfield BarFooConnector because then the hierarchy between what is a connector vs what is a desc isn't clear? this is pretty nitpicky so tioli!

Reply:
lol I spent literally 30m on exactly this. Calling this just `Source` is super weird for `sql::Source` and `catalog::Source` which both want to embed this field as _something_.

The best I came up with was calling this `SourceConfig`, which contained an `ExternalSourceConfig`, which contained an `ExternalSourceConnector`. And then you have `dataflow_types::SourceDesc`/`catalog::Source`/`sql::Source`, each of which contains a `SourceConfig`.

Reply:
we should have all been philosophy majors :D i think thats a substantial improvement